src/wcc/gen_wasm.c

```diff
@@ -1358,6 +1392,25 @@ static void gen_continue(void) {
   ADD_ULEB128(cur_depth - continue_depth - 1);
 }
 
+static void gen_goto(Stmt *stmt) {
+  const Name *label_name = stmt->goto_.label->ident;
+
+  // Create a patch record
+  GotoPatch *patch = malloc_or_die(sizeof(GotoPatch));
+  patch->label_name = label_name;
+  patch->goto_depth = cur_depth;
+   // Current position where we'll emit the branch depth
+  patch->patch_offset = CODE->len;
+  vec_push(goto_patches, patch);
+
+  // Generate the branch instruction with placeholder depth (will be patched later)
+  ADD_CODE(OP_BR);
+  ADD_ULEB128(0);
+
+  // Generate unreachable to handle WebAssembly stack polymorphism
+  ADD_CODE(OP_UNREACHABLE);
```

Comment:
Could you give me an example of where `unreachable` is needed?

Reply:
If you don't need to output `unreachable`, could you please remove this line?

Reply:
You are right, I don't think `unreachable` is needed for XCC, even if there are unreachable statements after the goto.